ui/lib/sync/addon/routes/secrets/destinations/index.ts

```diff
@@ -35,7 +49,7 @@ export default class SyncSecretsDestinationsIndexRoute extends Route {
   };
 
   redirect(model: ModelFrom<SyncSecretsDestinationsIndexRoute>) {
-    if (model.destinations.length === 0) {
+    if (!model.destinations.meta.total) {
       this.router.transitionTo('vault.cluster.sync.secrets.overview');
     }
   }
```

Comment:
Since a user can filter the results this could lead to 0 destinations returned from `lazyPaginatedQuery` but we only want to redirect if there are none returned from the server.